tracks/rust/exercises/circular-buffer/mentoring.md

```diff
@@ -0,0 +1,204 @@
+### Concepts
+
+- [Defining Error Types](https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/define_error_type.html)
+- [Trait Bounds](https://doc.rust-lang.org/rust-by-example/generics/bounds.html)
+
+### Reasonable Solutions
+
+A reasonable solution should:
+
+- Return an `Error` when appropriate
+- Limit the use of `unwrap`
+- Exhibit Separation of Concerns by defining additional helper functions to perform logically distinct operations
```

Comment:
Optionally add a link to "Separation of Concerns" for those mentors who are less-acquainted.